lib/ansible/modules/cloud/vmware/vmware_host_ntp.py

```diff
@@ -0,0 +1,265 @@
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+from ansible.module_utils._text import to_native
+
+
+class VmwareNtpConfigManager(PyVmomi):
+    def __init__(self, module):
+        super(VmwareNtpConfigManager, self).__init__(module)
+        cluster_name = self.params.get('cluster_name', None)
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        self.ntp_servers = self.params.get('ntp_servers', list())
+        self.hosts = []
+        if cluster_name:
+            cluster_obj = self.find_cluster_by_name(cluster_name=cluster_name)
+            if cluster_obj:
+                self.hosts = [host for host in cluster_obj.host]
+            else:
+                module.fail_json(changed=False, msg="Cluster '%s' not found" % cluster_name)
+        elif esxi_host_name:
+            esxi_host_obj = self.find_hostsystem_by_name(host_name=esxi_host_name)
+            if esxi_host_obj:
+                self.hosts = [esxi_host_obj]
+            else:
+                module.fail_json(changed=False, msg="ESXi '%s' not found" % esxi_host_name)
+        self.results = {}
+        self.desired_state = module.params['state']
+
+    def service_ctl(self, host, service_state):
+        changed = False
+        host_service_system = host.configManager.serviceSystem
+        if host_service_system:
+            try:
+                if service_state == 'start':
+                    host_service_system.StartService(id='ntpd')
+                if service_state == 'stop':
+                    host_service_system.StopService(id='ntpd')
+                if service_state == 'restart':
+                    host_service_system.RestartService(id='ntpd')
+                changed = True
+            except (vim.fault.InvalidState, vim.fault.NotFound, vim.fault.HostConfigFault) as e:
+                self.results[host.name]['error'] = e.msg
+
+        return changed
+
+    def update_ntp_settings(self, host, ntp_servers):
+        changed = False
+        host_date_time_manager = host.configManager.dateTimeSystem
+        if host_date_time_manager:
+            available_ntp_servers = host_date_time_manager.dateTimeInfo.ntpConfig.server
+            new_ntp_servers = available_ntp_servers + ntp_servers
+
+            ntp_config_spec = vim.host.NtpConfig()
+            ntp_config_spec.server = new_ntp_servers
+            date_config_spec = vim.host.DateTimeConfig()
+            date_config_spec.ntpConfig = ntp_config_spec
+            try:
+                host_date_time_manager.UpdateDateTimeConfig(date_config_spec)
+                changed = True
+            except vim.fault.HostConfigFault as e:
+                self.results[host.name]['error'] = e.msg
+            except Exception as e:
+                self.results[host.name]['error'] = e
+
+        return changed
+
+    def delete_ntp_settings(self, host, ntp_servers):
```

Comment:
@Akasurde Maybe you should rename "delete_ntp_settings" to "delete_ntp_servers".
What do you think ?

Reply:
OK.